src/cypress/support/commands.js

```diff
@@ -0,0 +1,173 @@
+  currency = symbol.replace('/', '')
+
+  cy.log('').then(() => {
+    cy.get('div[data-testid="tile-' + currency + '"] input')
+      .clear()
+      .type(notional)
+    cy.get(
+      'div[data-testid="tile-' +
+        currency +
+        '"] > div > div > div:nth-child(2)  > div:nth-child(2) > div > button[direction=' +
+        direction +
+        '] > div > div'
+    ).each((el, index, list) => {
+      if (index == 0) {
+        currentPrice = el.find('div:nth-child(2)').text()
+      } else {
+        currentPrice = currentPrice + el.text()
+      }
+    })
+  })
+
+  cy.log('').then(() => {
+    cy.log('current price is ' + currentPrice)
+    finalAmount = Number(currentPrice) * Number(notional)
+
+    cy.get(
+      'div[data-testid="tile-' +
+        currency +
+        '"] > div > div > div:nth-child(2)  > div:nth-child(2) > div > button[direction=' +
+        direction +
+        ']'
+    ).click({ force: true })
+    if (currency == 'EURJPY') {
+      cy.wait(2000)
+      cy.get('div[data-testid=tile-' + currency + '] div[role=dialog] > div[role=alert]').then(
+        el => {
+          cy.log('waiting text ' + el.text())
+          waitingText = el.text()
+
+          cy.wait(3000)
+        }
+      )
+    } else {
+      cy.wait(5000)
+    }
+    cy.get('div[data-testid=tile-' + currency + '] div[role=dialog] > div:nth-child(2)').then(
+      el => {
+        let tradeIDList = el.text().split(' ')
+        finalList.push(tradeIDList[2])
+      }
+    )
+    cy.get('div[data-testid=tile-' + currency + '] div[role=dialog] > div[role=alert]').then(el => {
+      cy.log('Confirmation text ' + el.text())
+      let confirmationText = el.text()
+      if (el.text().includes('rejected')) {
+        textList = confirmationText.split(' ')
+        finalList.push(textList[4])
+
+        finalList.push(currentYear)
+        finalList.push(direction)
+        finalList.push(currency)
+        if (direction.toLowerCase() == 'buy') {
+          finalList.push(currency.substring(0, 3))
+        } else {
+          finalList.push(currency.substring(3, 6))
+        }
+        finalList.push(notional)
+        finalList.push(currentPrice)
+        finalList.push(currentYear)
+        finalList.push('JPW')
+      } else {
+        textList = el.text().split(' ')
+        finalList.push('Done')
+        finalList.push(textList[14] + '-' + textList[15].replace('.', '') + '-' + currentYear)
+        finalList.push(direction)
+        finalList.push(textList[2] + textList[10])
+        if (direction.toLowerCase() == 'buy') {
+          finalList.push(textList[2])
+        } else {
+          finalList.push(textList[10])
+        }
+        finalList.push(textList[3])
+        finalList.push(textList[8])
+        finalList.push(textList[14] + '-' + textList[15].replace('.', '') + '-' + currentYear)
+        finalList.push('JPW')
+      }
+    })
+  })
+})
+
+Cypress.Commands.add('verifyTradeRejected', () => {
+  cy.log('').then(() => {
```

Comment:
An example of how to assert this without extracting bits into an ambiguous array in a previous command - 
```
Cypress.Commands.add('verifyTradeRejected', (symbol) => {
  cy.get(`[data-testid="tile-${symbol}"]`).contains('rejected')
}
```